lib/src/model/robotoff_nutrient_extraction.dart

```diff
@@ -0,0 +1,201 @@
+import 'package:json_annotation/json_annotation.dart';
+import '../interface/json_object.dart';
+
+part 'robotoff_nutrient_extraction.g.dart';
+
+@JsonSerializable()
+class RobotoffNutrientExtractionResult extends JsonObject {
+  final String? status;
+  final int? count;
+  final List<NutrientExtractionInsight>? insights;
+
+  const RobotoffNutrientExtractionResult({
+    this.status,
+    this.count,
+    this.insights,
+  });
+
+  NutrientExtractionInsight? get getLatestInsights {
+    insights?.sort((a, b) => a.completedAt!.compareTo(b.completedAt!));
+    return insights?.last;
+  }
+
+  NutrientEntity? getNutrientEntity(String nutrientKey) {
+    return getLatestInsights?.data?.nutrients?[nutrientKey];
+  }
+
+  factory RobotoffNutrientExtractionResult.fromJson(
+          Map<String, dynamic> json) =>
+      _$RobotoffNutrientExtractionResultFromJson(json);
+
+  @override
+  Map<String, dynamic> toJson() =>
+      _$RobotoffNutrientExtractionResultToJson(this);
+}
+
+@JsonSerializable()
+class NutrientEntity {
+  final int? start;
+  final int? end;
+  final String? text;
+  final String? unit;
+  final double? score;
+  final bool? valid;
+  final String? value;
+  final String? entity;
+  @JsonKey(name: 'char_start')
+  final int? charStart;
+  @JsonKey(name: 'char_end')
+  final int? charEnd;
+
+  const NutrientEntity({
+    this.start,
+    this.end,
+    this.text,
+    this.unit,
+    this.score,
+    this.valid,
+    this.value,
+    this.entity,
+    this.charStart,
+    this.charEnd,
+  });
+
+  factory NutrientEntity.fromJson(Map<String, dynamic> json) =>
+      _$NutrientEntityFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientEntityToJson(this);
+}
+
+@JsonSerializable()
+class NutrientAnnotationData {
+  final String? unit;
+  final String? value;
+
+  const NutrientAnnotationData({
+    this.unit,
+    this.value,
+  });
+
+  factory NutrientAnnotationData.fromJson(Map<String, dynamic> json) =>
+      _$NutrientAnnotationDataFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientAnnotationDataToJson(this);
+}
+
+@JsonSerializable()
+class NutrientAnnotation {
+  final Map<String, NutrientAnnotationData>? nutrients;
+  @JsonKey(name: 'serving_size')
+  final String? servingSize;
+  @JsonKey(name: 'nutrition_data_per')
+  final String? nutritionDataPer;
+
+  const NutrientAnnotation({
+    this.nutrients,
+    this.servingSize,
+    this.nutritionDataPer,
+  });
+
+  factory NutrientAnnotation.fromJson(Map<String, dynamic> json) =>
+      _$NutrientAnnotationFromJson(json);
+
+  Map<String, dynamic> toJson() => _$NutrientAnnotationToJson(this);
+}
+
+@JsonSerializable()
+class NutrientDataWrapper {
+  final Map<String, List<NutrientEntity>>? entities;
+  final Map<String, NutrientEntity>? nutrients;
```

Comment:
That `String` key is a vague.
Please consider adding getters with `Nutrient` and `PerSize` parameters.